device/tt_silicon_driver.cpp

```diff
@@ -1475,7 +1497,7 @@ bool tt_SiliconDevice::init_hugepage(chip_id_t device_id) {
         pin_pages.in.virtual_address = reinterpret_cast<std::uintptr_t>(mapping);
         pin_pages.in.size = hugepage_size;
 
-        auto &fd = g_SINGLE_PIN_PAGE_PER_FD_WORKAROND ? dev->device_fd_per_host_ch[ch] : dev->device_fd;
+        auto fd = dev->get_fd();
 
         if (ioctl(fd, TENSTORRENT_IOCTL_PIN_PAGES, &pin_pages) == -1) {
```

Comment:
I plan to move this to PCIDevice, so we encapsulate all ioctls there, and reduce code in future tt::umd::Chip. Just FYI, no action on this comment.